src/sdl1/controller.c

```diff
@@ -35,15 +35,15 @@ const uint8_t GAMEPAD_BUTTON_LSHOULDER = 4;
 const uint8_t GAMEPAD_BUTTON_RSHOULDER = 5;
 const uint8_t GAMEPAD_BUTTON_START = 11;
 const uint8_t GAMEPAD_BUTTON_BACK = 10;
-#elif defined(_3DS)
-const uint8_t GAMEPAD_BUTTON_DOWN = 8;
-const uint8_t GAMEPAD_BUTTON_LEFT = 9;
+#elif defined(__NDS__) || defined(_3DS)
+const uint8_t GAMEPAD_BUTTON_DOWN = 10;
+const uint8_t GAMEPAD_BUTTON_LEFT = 10;
 const uint8_t GAMEPAD_BUTTON_UP = 10;
-const uint8_t GAMEPAD_BUTTON_RIGHT = 11;
-const uint8_t GAMEPAD_BUTTON_A = 2;
-const uint8_t GAMEPAD_BUTTON_B = 1;
-const uint8_t GAMEPAD_BUTTON_X = 4;
-const uint8_t GAMEPAD_BUTTON_Y = 3;
+const uint8_t GAMEPAD_BUTTON_RIGHT = 10;
+const uint8_t GAMEPAD_BUTTON_A = 1;
+const uint8_t GAMEPAD_BUTTON_B = 2;
+const uint8_t GAMEPAD_BUTTON_X = 3;
+const uint8_t GAMEPAD_BUTTON_Y = 4;
```

Comment:
I think these changes will break the 3DS port as it is today, because right now it's built with an older version of SDL where the dpad buttons are buttons, and not a hat (which these changes assume).

Would you mind keeping NDS and 3DS separated? Basically this is what I have in mind:
1. Keep NDS and 3DS as separated `#elif defined(...)`
2. In the 3DS part, change the ABXY buttons as you suggest, but keep the dpad buttons as I had them before.
3. In the 3DS part, leave a comment like this `// TODO: When built with the latest SDL for 3DS, the dpad becomes a hat instead of 4 buttons. At that point 3DS and NDS will use the same button values`

Does that sound good? I'd rather not make another SDL update because things can get messy and I need time to fix whatever happens from that upgrade 😂